### PR TITLE
docs intro what neural search

### PR DESCRIPTION
**Pull request by alexcg1**
- docs(what-is-neural-search): rewriting
- docs(what-is-neural-search): style guide adherence, next step links

### All Submissions:

* [x] Have you followed the [documentation style guide](https://github.com/jina-ai/docs/tree/master/page_templates/style_guide.md)? 

### New Feature Submissions:

1. [x] Have you added your new file to a table of content index? 
2. [x] Did you use one of the templates in the [page_template](https://github.com/jina-ai/docs/tree/master/page_templates) folder?
3. [x] Is your title in Title Case? 
4. [x] Are your section headers in sentence case?

### CODE REVIEW

**Review by CatStark — APPROVED** [review]
LGTM

File: chapters/what_neural_search.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# What is Neural Search?
```
Comment (CatStark) [inline chapters/what_neural_search.md]: Just one thing, this should be in title case

**Comment by FionnD** [comment]
That was super quick @alexcg1 ! I won't get time until tomorrow to review if that's okay.

File: chapters/what_neural_search.md (hunk 1 above)
Comment (alexcg1) [reply chapters/what_neural_search.md]: Fixed!

**Review by CatStark — APPROVED** [review]

File: chapters/what_neural_search.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+- You want to understand **why** you're getting your *those* results
+- Your computing power is highly limited
+- You have a small amount of data (neural search requires big data)
+
+On the other hand, there are some situations where neural search makes more sense
+
+- Your search needs are likely to expand over different languages and data types
+- Accurate results are important, even if you can't perfectly understand why you got *those* results
+- You have a "noisy" set of data to search through
+- You have a large dataset for the neural network to "learn"
```
Comment (FionnD) [inline chapters/what_neural_search.md]: this is the only line I'd disagree with, with pre-trained models I think you can work with quite a small corpus. Even two or three hundred image + pre-trained image net would work.  What do you think?

**Review by FionnD — CHANGES_REQUESTED** [review]
Really good! and very valuable, I would even put it not front page as a link? 

Only one comment

File: chapters/what_neural_search.md (hunk 2 above)
Comment (alexcg1) [reply chapters/what_neural_search.md]: Yup. This section was adapted from my old version which was adapted from Han's original version. I think we've seen it do well in action so I'll remove that bullet point

**Comment by alexcg1** [comment]
> Really good! and very valuable, I would even put it not front page as a link?
> 
> Only one comment

Even put it not front page as a link? You mean on front page?

File: chapters/what_neural_search.md (hunk 2 above)
Comment (alexcg1) [reply chapters/what_neural_search.md]: Resolved

**Comment by FionnD** [comment]
> > Really good! and very valuable, I would even put it not front page as a link?
> > Only one comment
> 
> Even put it not front page as a link? You mean on front page?

Yes, English failed me after a beer. Let's do what you said.

**Comment by alexcg1** [comment]
@FionnD can you approve the changes? Then I'll merge

**Review by FionnD — APPROVED** [review]

**Pull request merged**